### DISCUSSION TITLE
How to overlook The content-type of the response we received from your web server was: undefined error while cy.visit()?

### DISCUSSION THREAD

**Discussion by pranesh517** [cypress-io/cypress#28652] [Questions and Help]
Is there any way to cypress to overlook content-type: undefined error while cy.visit()?
Tried using 
cy.visit({
            url: 'url',
            failOnStatusCode: false,
            headers: {'Accept': 'text/html'}
        });

<img width="380" alt="image" src="https://github.com/cypress-io/cypress/assets/39439293/75c3646b-5224-4343-bd03-8d9276c3ccb0">

**Comment by portyu9** [comment]
Hi @pranesh517 

You cannot bypass this requirement with `failOnStatusCode: false`.

`cy.visit()` requires the final response to have an HTML content type:

```text
Content-Type: text/html
```

`failOnStatusCode` only controls whether Cypress fails because of the HTTP **status code**. It does not disable the `Content-Type` requirement.

Also, setting:

```js
headers: {
  Accept: 'text/html'
}
```

only changes the **request** header. It does not make the server send a `Content-Type` response header.

The correct fix is therefore on the server/reverse-proxy/redirect target: make the page response include something such as:

```text
Content-Type: text/html; charset=utf-8
```

If you are not actually trying to load a browser page and only want to check that the URL responds successfully, use `cy.request()` instead of `cy.visit()`.

Please mark this answer as accepted if it helped, thank you.
